src/main/java/com/google/sps/servlets/ActionsServlet.java

```diff
@@ -49,91 +44,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   private List<Project> extractProjectsList(String originalJsonString) {
     Gson gson = new Gson();
-    Map jsonMap = gson.fromJson(originalJsonString, Map.class);
+    Type mapType = new TypeToken<Map<String, Object>>() {}.getType();
+    Map<String, Object> jsonMap = gson.fromJson(originalJsonString, mapType);
     String[] jsonSections = {"search", "response", "projects"};
     for (String section : jsonSections) {
-      jsonMap = (Map) jsonMap.get(section);
+      jsonMap = (Map<String, Object>) jsonMap.get(section);
     }
-    List projectsListJson = (List) jsonMap.get("project");
+    List<Map<String, Object>> projectsListJson = (List<Map<String, Object>>) jsonMap.get("project");
     List<Project> projectsList = new ArrayList<>();
     for (int i = 0; i < projectsListJson.size(); i++) {
-      Map projectJson = (Map) projectsListJson.get(i);
-      double id = (double) projectJson.get("id");
-      String title = (String) projectJson.get("title");
-      String summary = (String) projectJson.get("summary");
-      Project project = new Project.ProjectBuilder(id)
-                                  .withTitle(title)
-                                  .withSummary(summary)
-                                  .build();
+      Map<String, Object> projectJson = projectsListJson.get(i);
+      Project project = getProjectFromMap(projectJson);
       projectsList.add(project);
     }
     return projectsList;
   }
 
-  /**
-  * MIT License
-  * 
-  * Copyright (c) 2017 Eugen Paraschiv
-  * 
-  * Permission is hereby granted, free of charge, to any person obtaining a copy
-  * of this software and associated documentation files (the "Software"), to deal
-  * in the Software without restriction, including without limitation the rights
-  * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-  * copies of the Software, and to permit persons to whom the Software is
-  * furnished to do so, subject to the following conditions:
-  * 
-  * The above copyright notice and this permission notice shall be included in all
-  * copies or substantial portions of the Software.
-  */
-
-  private String getParamsString(Map<String, String> params) throws UnsupportedEncodingException {
-      StringBuilder result = new StringBuilder();
-
-      for (Map.Entry<String, String> entry : params.entrySet()) {
-          result.append(URLEncoder.encode(entry.getKey(), "UTF-8"));
-          result.append("=");
-          result.append(URLEncoder.encode(entry.getValue(), "UTF-8"));
-          result.append("&");
-      }
-
-      String resultString = result.toString();
-      return resultString.length() > 0 ? resultString.substring(0, resultString.length() - 1) : resultString;
+  private Project getProjectFromMap(Map projectMap) {
+    double id = (double) projectMap.get("id");
+    String title = (String) projectMap.get("title");
+    String summary = (String) projectMap.get("summary");
+    Project project = new Project.ProjectBuilder(id)
```

Comment:
When chaining methods the next line is supposed to be indented by 4 spaces for the beginning of the previous line
```
CoolObject myObject = new CoolObject.CoolObjectBuilder(requiredParam)
    .withOptionalParam(optionalParam)
```